From e5d9edc147c1c1a3be030ee47de89daf6953beb2 Mon Sep 17 00:00:00 2001
From: Raphael Geissert <geissert@debian.org>
Date: Thu, 15 Dec 2016 16:46:11 +0100
Subject: [PATCH] apt.systemd.daily: call unattended-upgrade --fix-only

Drop the call to apt-get check as it prevents apt-daily from working
in case the dpkg database is in a dirty state.  The check was meant to
detect and exit in case the lock was being held by another process,
but it would offer no guarantee later on.

Replace it with a call to u-u --fix-only, a new option, to repair the
dpkg database if needed.

Closes: #838369
---
 debian/apt.systemd.daily |   14 ++++++++++----
 1 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/debian/apt.systemd.daily b/debian/apt.systemd.daily
index 03d41ba..52004f5 100755
--- a/debian/apt.systemd.daily
+++ b/debian/apt.systemd.daily
@@ -337,10 +337,16 @@ if [ "$VERBOSE" -ge 3 ]; then
     set -x
 fi
 
-# check if we can lock the cache and if the cache is clean
-if which apt-get >/dev/null 2>&1 && ! eval apt-get check $XAPTOPT $XSTDERR ; then
-    debug_echo "error encountered in cron job with \"apt-get check\"."
-    exit 0
+# call unattended-upgrade to fix the dpkg database if needed
+if which unattended-upgrade >/dev/null 2>&1 && ! eval unattended-upgrade --fix-only $XSTDOUT $XSTDERR ; then
+    exit_code=$?
+
+    # if u-u doesn't know about --fix-only ignore the error
+    # (OptionParser exits with code 2 in that case)
+    if [ $exit_code -ne 2 ]; then
+	debug_echo "error encountered in cron job with \"unattended-upgrade --fix-only\"."
+	exit $exit_code
+    fi
 fi
 
 # Global current time in seconds since 1970-01-01 00:00:00 UTC
-- 
1.7.2.5

